swarm_dial.go

```diff
@@ -249,8 +296,10 @@ func (s *Swarm) dial(ctx context.Context, p peer.ID) (*Conn, error) {
 	*/
 	paddrs := s.peers.Addrs(p)
 	goodAddrs := addrutil.FilterAddrs(paddrs,
-		addrutil.AddrUsableFunc,
 		subtractFilter,
+		s.canDial,
+		// TODO: Consider allowing this?
```

Comment:
Eventually, I'd like to allow link-local stuff. For now, this preserves existing behaviour.